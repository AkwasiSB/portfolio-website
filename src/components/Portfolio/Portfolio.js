import React from 'react';
import styles from './Portfolio.scss';

export default class Portfolio extends React.Component {
	constructor(props) {
		super(props);

		this.projects = [{
			id: 1,
			title: 'Bilingua',
			image: require('../../images/bilingua.png'),
			link: 'https://bilingua.io',
			details: 'A language exchange app designed to help people from different nationalities to easily learn languages of their interest through social interaction.',
		},{
			id: 2,
			title: 'Transwap',
			image: require('../../images/transwap.png'),
			link: 'https://transwap.com',
			details: 'A remittance website for money transfers in South-East asian countries.',
		},{
			id: 3,
			title: 'Imaginae',
			image: require('../../images/imaginae.png'),
			link: 'https://imaginae.fr',
			details: 'A web and mobile app for creating a community of authors and readers especially the French language.',
		}];
	}

	renderProjects = () => {
		return this.projects.map((project) => (
			<a href={project.link} target='_blank' rel='noopener noreferrer' key={project.id} className={styles.project}>
				<img src={project.image} alt='' />

				<h4>{project.title}</h4>

				<div className={styles.projectDetails}>
					<h4>{project.title}</h4>

					<p>{project.details}</p>
				</div>
			</a>
		));
	}

	render() {
		return (
			<section id='portfolio' className={styles.portfolio}>
				<h2>PORTFOLIO</h2>

				<div className={styles.portfolioContainer}>
					{this.renderProjects()}
				</div>
			</section>
		);
	}
}
