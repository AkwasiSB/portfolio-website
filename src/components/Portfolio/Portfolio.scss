$screen-sm: 768px;
$screen-md: 1200px;

.portfolio {
	display: flex;
	flex-direction: column;
	padding: 6rem 2rem 2rem;

	@media (min-width: $screen-sm) {
		h2 {
			padding-left: 13%;
		}
	}
}

.portfolio-container {
	display: flex;
	justify-content: center;
	align-items: center;
	flex-wrap: wrap;
	padding: 2rem 1rem;
}

.project {
	position: relative;
	display: flex;
	flex-direction: column;
	align-items: center;
	margin: 2rem 0;
	width: 30rem;
	height: 30rem;
	border-radius: 4rem;
	box-shadow: 0 3px 15px 0 rgba(0, 0, 0, 0.25);

	img {
		width: 30rem;
		border-radius: 2rem;
		margin-top: 0;
		transition-property: margin, width;
		transition-duration: 0.5s;
		transition-timing-function: ease-in;

		@media (min-width: $screen-sm)  {
			width: 25rem;
			margin-top: 1.5rem;
		}
	}

	& > h4 {
		opacity: 1;
		color: #555;
		transition: opacity 0.5s ease-in;
		display: none;

		@media (min-width: $screen-sm)  {
			display: block;
		}
	}

	&-details {
		display: flex;
		flex-direction: column;
		align-items: center;
		width: 300px;
		color: #555;
		background-color: #fff;
		line-height: 1.8rem;
		border-bottom-left-radius: 4rem;
		border-bottom-right-radius: 4rem;

		h4 {
			margin-bottom: 0;
		}

		p {
			font-size: 1.4rem;
			padding: 0 1.5rem;
		}

		@media (min-width: $screen-sm)  {
			position: absolute;
			top: 100%;
			bottom: 0;
			left: 0;
			right: 0;
			font-size: 1.2rem;
			color: #fff;
			background-color: rgba(3, 3, 3, 0.6);
			width: 100%;
			overflow: hidden;
			border-bottom-left-radius: 4rem;
			border-bottom-right-radius: 4rem;
			transition: top 0.5s ease-in;
		}
	}

	@media (min-width: $screen-sm)  {
		height: 20rem;
		border: 1px solid #ccc;
		margin-left: 2rem;
		margin-right: 2rem;

		&:hover img {
			width: 30rem;
			margin-top: 0;
		}

		&:hover > h4 {
			opacity: 0;
		}

		&:hover &-details {
			top: 32%;
		}
	}
}
