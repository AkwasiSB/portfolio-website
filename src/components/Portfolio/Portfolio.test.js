import Portfolio from './Portfolio';

it('should render <Portfolio /> correctly', () => {
    const portfolioWrapper = shallow(<Portfolio />);

    expect(portfolioWrapper).toMatchSnapshot();
});
