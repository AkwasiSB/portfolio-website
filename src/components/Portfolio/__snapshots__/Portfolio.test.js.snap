// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Portfolio /> correctly 1`] = `
<section
  className="portfolio"
  id="portfolio"
>
  <h2>
    PORTFOLIO
  </h2>
  <div
    className="portfolioContainer"
  >
    <a
      className="project"
      href="https://bilingua.io"
      key="1"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt=""
        src={
          Object {
            "default": "",
          }
        }
      />
      <h4>
        Bilingua
      </h4>
      <div
        className="projectDetails"
      >
        <h4>
          Bilingua
        </h4>
        <p>
          A language exchange app designed to help people from different nationalities to easily learn languages of their interest through social interaction.
        </p>
      </div>
    </a>
    <a
      className="project"
      href="https://transwap.com"
      key="2"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt=""
        src={
          Object {
            "default": "",
          }
        }
      />
      <h4>
        Transwap
      </h4>
      <div
        className="projectDetails"
      >
        <h4>
          Transwap
        </h4>
        <p>
          A remittance website for money transfers in South-East asian countries.
        </p>
      </div>
    </a>
    <a
      className="project"
      href="https://imaginae.fr"
      key="3"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt=""
        src={
          Object {
            "default": "",
          }
        }
      />
      <h4>
        Imaginae
      </h4>
      <div
        className="projectDetails"
      >
        <h4>
          Imaginae
        </h4>
        <p>
          A web and mobile app for creating a community of authors and readers especially the French language.
        </p>
      </div>
    </a>
  </div>
</section>
`;
